In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/raceInProgress/raceV3.csv')

In [3]:
df.head()

PcrKey  black  white  hispanic/latino  asian  \
0  13273412      0      1                0      0   
1  17184298      0      1                0      0   
2  21699657      0      1                0      0   
3  72057010      0      1                0      0   
4  83733589      0      0                0      0   

   americanIndian/alaskaNative  nativeHawaiian/otherPacificIslander  
0                            0                                    0  
1                            0                                    0  
2                            0                                    0  
3                            0                                    0  
4                            1                                    0

- this is now clean as all hell

In [8]:
len(df.index)

32866360

- Total Sample Size = 32,866,360
- Next step is to import this into sql db, then join the computed elements

In [12]:
schema = pd.io.sql.get_schema(df, 'raceTable')
schema = schema.replace("\n", "")
schema = schema.replace('"','')
schema

'CREATE TABLE raceTable (PcrKey INTEGER,  black INTEGER,  white INTEGER,  hispanic/latino INTEGER,  asian INTEGER,  americanIndian/alaskaNative INTEGER,  nativeHawaiian/otherPacificIslander INTEGER)'

In [13]:
# cols list for insert statement
cols = list(df.columns)
print(',\n'.join(map(str, cols)))

PcrKey,
black,
white,
hispanic/latino,
asian,
americanIndian/alaskaNative,
nativeHawaiian/otherPacificIslander


In [14]:
len(cols)

7

In [15]:
q = [' ?'] * 7
print(','.join(map(str, q)))

 ?, ?, ?, ?, ?, ?, ?


- One little fix, SQL does not like that I used slashes in the column names, lets fix that

In [16]:
df.head()

PcrKey  black  white  hispanic/latino  asian  \
0  13273412      0      1                0      0   
1  17184298      0      1                0      0   
2  21699657      0      1                0      0   
3  72057010      0      1                0      0   
4  83733589      0      0                0      0   

   americanIndian/alaskaNative  nativeHawaiian/otherPacificIslander  
0                            0                                    0  
1                            0                                    0  
2                            0                                    0  
3                            0                                    0  
4                            1                                    0

In [18]:
df.rename(columns = {'hispanic/latino':'hispanic_latino', 'americanIndian/alaskaNative':'americanIndian_alaskaNative', 'nativeHawaiian/otherPacificIslander':'nativeHawaiian_otherPacificIslander'}, inplace = True)

In [19]:
df.head()

PcrKey  black  white  hispanic_latino  asian  \
0  13273412      0      1                0      0   
1  17184298      0      1                0      0   
2  21699657      0      1                0      0   
3  72057010      0      1                0      0   
4  83733589      0      0                0      0   

   americanIndian_alaskaNative  nativeHawaiian_otherPacificIslander  
0                            0                                    0  
1                            0                                    0  
2                            0                                    0  
3                            0                                    0  
4                            1                                    0

In [20]:
df.to_csv('/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/raceInProgress/raceV4.csv',index=False)